src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -833,6 +833,46 @@ void ObjectCopyRequest<I>::compute_dst_object_may_exist() {
                    << dendl;
 }
 
+template <typename I>
+void ObjectCopyRequest<I>::trigger_copyup() {
+  // With mirroring, a write to a non-existent primary cloned image object will 
+  // trigger a copyup which updates the older synced snaps with the parent data.
+  // This causes the rbd-mirror to ignore the parent data and only sync the new write
+  // to the clone, causing data corruption.
+  // For rbd-mirror scenarios, we trigger a copyup on the object and write the parent
+  // data to the object before writing the new clone image data.
```

Comment:
nit: I would not use rbd-mirror/mirroring words here, it may be any other application or tool that is doing deep copy for a clone. BTW, I have not checked but I suppose the `rbd deep-copy` could be also extended to provide similar functionality.

Reply:
Maybe I can clarify that further in the comment. The issue is seen only with rbd-mirroring as of now. The regular deep copy/migration etc work correctly.

Reply:
It is not about clarifying. It is about mentioning a specific application (rbd-mirror) in librbd code. Just describe the scenario but don't specify it is "rbd-mirror" scenario. Actually I am not sure we need this comment at all, but I would leave this to Ilya to decide.

Reply:
I believe this comment is required to help future devs understand what is going on. The lack of comments in the code makes it difficult to figure out why something is being done.
I have rewritten it to remove mention of rbd-mirror, however I personally think we need to have something written up as to which components use these functions and how.